src/main/scala/com/fulcrumgenomics/util/PhredValue.scala

```diff
@@ -0,0 +1,132 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import dagr.commons.CommonsDef._
+
+/** Methods and values to help convert to and from Phred-scaled probabilities.
+  *
+  * A number of implicit conversions are defined in this object.  Integers are implicitly are interpreted as
+  * literal phred values, while Floats and Doubles converted to Phred values using `toPhred`.
+  */
+object PhredValue {
+  /** Phred-scaled zero probability. */
+  val ZeroProbability = PhredValue(Double.PositiveInfinity)
+  /** Phred-scaled probability of one. */
+  val OneProbability  = PhredValue(0.0)
+  /** Phred value of 0. */
+  val ZeroPhred       = PhredValue(0.0)
+  /** The precision to use when converting to an Integer. */
+  val Precision       = 0.0000000001 // Q100
+  /** converts a probability to a phred score, with no validation. */
+  def toPhred(pr: Double) = -10.0*Math.log10(pr)
+  /** converts to a probability from a phred score, with no validation. */
+  def toPr(phred: Double) = Math.pow(10.0, phred / -10.0)
+  /** Gets the mean of the given phred values. */
+  def mean(phred: PhredValue*): PhredValue = phred.foldLeft(ZeroProbability)((sum, qual) => sum + qual) / phred.length.toDouble
+  /*
+  implicit def phredValueWrapper[T](v: T)(implicit numeric: Numeric[T]): PhredValue = new PhredValue(value=numeric.toDouble(v))
+  implicit class NumericPhredValue[T](v: T)(implicit numeric: Numeric[T]) {
+    def toPhredValue: PhredValue = new PhredValue(value=numeric.toDouble(v))
+  }
+  */
+  implicit def intToPhredWrapper(v: Int): PhredValue = new PhredValue(value=v)
+  implicit def floatToPhredWrapper(v: Float): PhredValue = {
+    //if (v < 0 || v > 1) throw new IllegalArgumentException(s"Float value is out of range to convert to phred: $v")
+    PhredValue(toPhred(v))
+  }
+  implicit def doubleToPhredWrapper(v: Double): PhredValue = {
+    //if (v < 0 || v > 1) throw new IllegalArgumentException(s"Double value is out of range to convert to phred: $v")
+    PhredValue(toPhred(v))
+  }
+  implicit class IntPhredValue(v: Int) {
+    def toPhredValue: PhredValue = new PhredValue(value=v.toDouble)
+  }
+  implicit class FloatPhredValue(v: Float) {
+    def toPhredValue: PhredValue = {
+      //if (v < 0 || v > 1) throw new IllegalArgumentException(s"Float value is out of range to convert to phred: $v")
+      new PhredValue(value=toPhred(v.toDouble))
+    }
+  }
+  implicit class DoublePhredValue(v: Double) {
+    def toPhredValue: PhredValue = {
+      //if (v < 0 || v > 1) throw new IllegalArgumentException(s"Double value is out of range to convert to phred: $v")
+      new PhredValue(value=toPhred(v.toDouble))
+    }
+  }
+}
+
+/** Represents Phred-scaled probabilities, which are `-10 * log10(Pr)`. */
+case class PhredValue(var value: Double) extends Ordered[PhredValue] {
```

Comment:
Yikes - I was assuming that PhredValue was immutable.  I think it should probably be immutable and also declared as a value class (see: http://docs.scala-lang.org/overviews/core/value-classes.html).  This will have the same performance semantics, but make it much easier to reason about PhredValues as they'll behave like other value types (Ints, Doubles).